account_operating_unit/models/invoice.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# - Jordi Ballester Alomar
+# © 2016 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from openerp import api, fields, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    operating_unit_id = fields.Many2one('operating.unit', 'Operating Unit',
+                                        default=lambda self:
+                                        self.env['res.users'].
+                                        operating_unit_default_get(self._uid))
+
+    @api.multi
+    def finalize_invoice_move_lines(self, move_lines):
+        move_lines = super(AccountInvoice,
+                           self).finalize_invoice_move_lines(move_lines)
+        new_move_lines = []
+        for line_tuple in move_lines:
+            if self.operating_unit_id:
+                line_tuple[2]['operating_unit_id'] = \
+                    self.operating_unit_id.id
+            new_move_lines.append(line_tuple)
+        return new_move_lines
+
+    @api.multi
+    def _check_company_operating_unit(self):
+        for pr in self.browse():
+            if (
+                pr.company_id and
+                pr.operating_unit_id and
+                pr.company_id != pr.operating_unit_id.company_id
+            ):
+                return False
+        return True
+
+    _constraints = [
+        (_check_company_operating_unit,
```

Comment:
Remove: the methods already has the `@constrains` decorator, and I don't need/shouldn't have it here.